File: day01/typescript/solution.ts
import { readFileSync } from "node:fs";

function parse(raw: string): string[] {
  return raw.trimEnd().split("\n");
}

function part1(_data: string[]): string | number {
  return "TODO";
}

function part2(_data: string[]): string | number {
  return "TODO";
}

const raw = readFileSync(0, "utf8");
const data = parse(raw);
console.log(`Part 1: ${part1(data)}`);
console.log(`Part 2: ${part2(data)}`);

File: day01/haskell/Solution.hs
module Main where

import Data.List (lines)

parse :: String -> [String]
parse = lines

part1 :: [String] -> String
part1 _ = "TODO"

part2 :: [String] -> String
part2 _ = "TODO"

main :: IO ()
main = do
  input <- getContents
  let d = parse input
  putStrLn $ "Part 1: " ++ part1 d
  putStrLn $ "Part 2: " ++ part2 d
